src/System.Management.Automation/commands/utility/FormatAndOutput/out-console/OutConsole.cs

```diff
@@ -102,6 +98,7 @@ protected override void ProcessRecord()
         {
             if (Transcript)
             {
+                Host.UI.TranscribeOnly = true;
```

Comment:
I only learned that `Out-Default` had a `-Transcript` switch when this issues was opened and I started looking into it (documentation and tests are also lacking).

Best as I can tell, although you can have nested `Start-Transcript`, `Out-Default` is only used at the end of a pipeline, so the test case of `Out-Default -Transcript | Out-Default -Transcript` I don't think represents a real use case as the intent of `-Transcript` (as far as I can tell) is to suppress output to the pipeline and only send it to the transcript.  However, it got the console stuck in transcript mode, so I thought it was worth fixing (the other issue with exception being raised is a real use case).

So I don't think we need a counter here as it should really just toggle from `true` and `false` when this cmdlet is used.